packages/cozy-client/src/associations/HasManyFiles.js

```diff
@@ -3,12 +3,27 @@ import HasMany from './HasMany'
 import { QueryDefinition, Mutations } from '../queries/dsl'
 
 export default class HasManyFiles extends HasMany {
-  async fetchMore() {
-    const skip = this.getRelationship().data.length
+  async fetchMore(withCursor) {
     const queryDef = new QueryDefinition({ doctype: 'io.cozy.files' })
-    const response = await this.query(
-      queryDef.referencedBy(this.target).offset(skip)
-    )
+    const relationships = this.getRelationship().data
+    let response
+    if (withCursor) {
```

Comment:
If queryDef.sort is not defined, let's use cursors ?

Reply:
@edas I'm not a big fan of this kind of magic, but ok if you think it's better this way.
@ptbrowne No, sort is used for mango query, not for views. Also, how do you access the queryDef of the previous query from fetchMore?

Reply:
The management of skip was done automatically, looking at the state of the relationship to automatically add the skip/limit. I think it should be the same.

> Also, how do you access the queryDef of the previous query from fetchMore?

I think for the relationship to be able to access the part of the queryDefinition that is relevant to itself is expected. It may not be possible at the moment, but it would not disturb me to have

```
/** Returns the options from the parent query related to the relationship */
relationship.getQueryDefinitionPart()
```

For now, the DSL for relationship is very simple and does not allow to pass options but we could have:

```
// simple case, already supported
client.all('io.cozy.albums').include(['io.cozy.files'])

// more complex case where we need the files to be returned sorted
client.all('io.cozy.albums').include([{ doctype: 'io.cozy.files', sort: 'datime' ])
```